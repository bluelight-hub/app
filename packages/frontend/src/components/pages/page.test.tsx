import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { vi } from 'vitest';
import IndexPage from './page';

// Mock for useNavigate
const mockNavigate = vi.fn();
vi.mock('react-router', () => ({
    useNavigate: () => mockNavigate,
}));

describe('IndexPage', () => {
    // Reset mocks before each test
    beforeEach(() => {
        mockNavigate.mockReset();
    });

    // Unit test: Component renders correctly
    test('renders correctly with login button', () => {
        render(
            <BrowserRouter>
                <IndexPage />
            </BrowserRouter>
        );

        // Check if the button exists
        const loginButton = screen.getByText('Anmelden');
        expect(loginButton).toBeInTheDocument();
    });

    // Integration test: Button interaction works correctly
    test('navigates to /app when button is clicked', () => {
        render(
            <BrowserRouter>
                <IndexPage />
            </BrowserRouter>
        );

        // Find the button and click it
        const loginButton = screen.getByText('Anmelden');
        fireEvent.click(loginButton);

        // Verify navigation was called with correct path
        expect(mockNavigate).toHaveBeenCalledTimes(1);
        expect(mockNavigate).toHaveBeenCalledWith('/app');
    });

    // Snapshot test
    test('matches snapshot', () => {
        const { container } = render(
            <BrowserRouter>
                <IndexPage />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot();
    });
}); 