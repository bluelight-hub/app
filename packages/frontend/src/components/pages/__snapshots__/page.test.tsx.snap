// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IndexPage > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center h-screen w-screen"
  >
    <button
      class="ant-btn css-dev-only-do-not-override-19lec04 ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
      type="button"
    >
      <span>
        Anmelden
      </span>
    </button>
  </div>
</div>
`;
